presto-hive/src/main/java/com/facebook/presto/hive/HiveConnector.java

```diff
@@ -154,21 +154,21 @@ public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel
         checkConnectorSupports(READ_UNCOMMITTED, isolationLevel);
         ConnectorTransactionHandle transaction = new HiveTransactionHandle();
         try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
-            transactions.put(transaction, metadataFactory.create());
+            transactionManager.put(transaction, metadataFactory.create());
         }
         return transaction;
     }
 
     @Override
     public void commit(ConnectorTransactionHandle transaction)
     {
-        checkArgument(transactions.remove(transaction) != null, "no such transaction: %s", transaction);
+        checkArgument(transactionManager.remove(transaction) != null, "no such transaction: %s", transaction);
     }
 
     @Override
     public void rollback(ConnectorTransactionHandle transaction)
     {
-        HiveMetadata metadata = transactions.remove(transaction);
+        HiveMetadata metadata = (HiveMetadata) transactionManager.remove(transaction);
```

Comment:
You could move the following code to be inside of HiveTransactionManager, and then the HiveTransactionManager interface could become:
```
void beginTransaction(HiveTransactionHandle);
HiveMetadata getMetadata(HiveTransactionHandle)
void rollbackTransaction(HiveTransactionHandle);